setup.py

```diff
@@ -0,0 +1,231 @@
+        with open(self.license_file, "w") as f:
+            f.write(self.nvfuser_license_txt)
+
+
+try:
+    from wheel.bdist_wheel import bdist_wheel
+except ImportError:
+    build_whl = None
+else:
+
+    class build_whl(bdist_wheel):
+        def run(self):
+            with concat_third_party_license() as tp_licenses:
+
+                if len(tp_licenses) != 0:
+                    with open("LICENSE", "a") as f:
+                        f.write("\n\n")
+                        f.write(
+                            "NVIDIA/fuser depends on libraries with license listed below:"
+                        )
+
+                for project_name, license_files in tp_licenses.items():
+                    # check all license files are identical
+                    with open(license_files[0], "r") as f:
+                        license_ref = f.read()
+
+                    def check_file(file_name):
+                        with open(file_name, "r") as f:
+                            return f.read() == license_ref
+
+                    identical_flag = all(map(check_file, license_files[1:]))
+                    if not identical_flag:
+                        raise RuntimeError(
+                            "inconsistent license found for project: ",
+                            project_name,
+                            " check its license files under: ",
+                            license_files,
+                        )
+
+                    with open("LICENSE", "a") as f:
+                        f.write("\n\nProject Name: " + project_name)
+                        f.write("\nLicense Files:\n")
+                        for file_name in license_files:
+                            f.write("\t" + file_name)
+                        f.write("\n" + license_ref)
+
+                # generate whl before we restore LICENSE
+                super().run()
+
+
+def cmake():
+    # make build directories
+    build_dir_name = "build"
+    cwd = os.path.dirname(os.path.abspath(__file__))
+    cmake_build_dir = os.path.join(cwd, "build")
+    if not os.path.exists(cmake_build_dir):
+        os.makedirs(cmake_build_dir)
+
+    from tools.gen_nvfuser_version import get_pytorch_cmake_prefix
+
+    pytorch_cmake_config = "-DCMAKE_PREFIX_PATH=" + get_pytorch_cmake_prefix()
+
+    # generate cmake directory
+    cmd_str = [get_cmake_bin(), pytorch_cmake_config, "-B", build_dir_name, "."]
+    if not NO_TEST:
+        cmd_str.append("-DBUILD_TEST=ON")
+    if not NO_PYTHON:
+        cmd_str.append("-DBUILD_PYTHON=ON")
+    if not NO_BENCHMARK:
+        cmd_str.append("-DBUILD_NVFUSER_BENCHMARK=ON")
+
+    subprocess.check_call(cmd_str)
+
+    if not CMAKE_ONLY:
+        # build binary
+        max_jobs = str(multiprocessing.cpu_count())
+        cmd_str = [
+            get_cmake_bin(),
+            "--build",
+            build_dir_name,
+            "--target",
+            "install",
+            "--",
+            "-j",
+            max_jobs,
+        ]
```

Comment:
optionally will be better if we can set `MAX_JOBS` via an environment variable like https://github.com/pytorch/pytorch/blob/96e3b3ac724c4e262dc8fc05da18ebcbda6473b8/setup.py#L11-L12